operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java

```diff
@@ -117,7 +136,7 @@ public static Labels userLabels(Map<String, String> userLabels) {
         }
 
         // Remove Kubernetes Domain specific labels
-        Map<String, String> filteredLabels = userLabels
+        Map<String, String> filteredLabels = additionalLabels
```

Comment:
So I guess this is where it runs into https://github.com/strimzi/strimzi-kafka-operator/issues/2743 and it is not clear to me how to deal with it.

Reply:
For the short term I think we should modify this filter to allow a few whitelisted labels to be applied
https://github.com/strimzi/strimzi-kafka-operator/pull/2730/files#diff-6de62e326f52e6cc92e7ffd7e722e8f4R142

I suggest we only allow the `app.kubernetes.io/part-of` label as this one is used to group resources together? Looking at the others it is unclear what value it would have to allow them to be modifiable.

Reply:
That sounds reasonable - I have only `app.kubernetes.io/part-of` so far reported as an issue.